insights/parsers/openstack_server_list_all.py

```diff
@@ -0,0 +1,64 @@
+"""
+openstack server list --all command - Command
+=============================================
+Parsing and extracting data from output of command ``/usr/bin/openstack server list --all``.
+--------------------------------------------------------------------------------------------
+"""
+
+from insights import CommandParser, parser, LegacyItemAccess
+from insights.parsers import ParseException, SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.openstack_server_list_all)
+class OpenstackServerListAll(LegacyItemAccess, CommandParser):
+    """
+    The output of command ``/usr/bin/openstack server list --all`` is in table format,
+    contains 'Variable_name' and 'Value' two columns.
+    This parser will parse the table and set each variable as an class
+    attribute. The unparsable lines are stored in the ``bad_lines`` property list.
+    Sample command output::
+
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+        | ID                                   | Name         | Status | Networks               | Image          | Flavor     |
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+        | 410b05bb-59b7-4b4c-88e9-975c811d68da | compute-0    | ACTIVE | ctlplane=192.168.24.20 | overcloud-full | compute    |
+        | f891e98b-4df6-4c90-9bf1-39cf8ac900b0 | compute-1    | ACTIVE | ctlplane=192.168.24.9  | overcloud-full | compute    |
+        | 3d62cd7e-41d2-43dd-a5bf-5935bc319fae | controller-0 | ACTIVE | ctlplane=192.168.24.10 | overcloud-full | controller |
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+    """
+
+    def parse_content(self, content):
+        """
+        Parse output content table of command ``/usr/bin/openstack server list --all``.
+        Set each variable as an class attribute.
+        """
+        data = []
+        bad_lines = []
+        if not content:
+            raise SkipException("Empty content.")
+        if len(content) < 5:
+            raise ParseException("Wrong content in table: '{0}'.".format(content))
+
+        d = {}
+        for _l in content[3:-1]:
+            l = _l.strip()
+            if not (l.startswith('|') and l.endswith('|')):
+                bad_lines.append(_l)
+                continue
+            l = l[1:-1]
+            lsp = l.split('|', 5)
+            if len(lsp) < 6:
+                bad_lines.append(_l)
+                continue
+            d = {
+                "ID": lsp[0].strip(),
+                "Name": lsp[1].strip(),
+                "Status": lsp[2].strip(),
+                "Networks": lsp[3].strip(),
+                "Image": lsp[4].strip(),
+                "Flavor": lsp[5].strip()
+            }
+            data.append(d)
+            self.data = data
```

Comment:
Although `data` is used in a lot of parsers, its not the most accurate attribute name.  Wouldn't `servers` be a more appropriate attribute name here?